dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/PageInfo.java

```diff
@@ -65,10 +65,17 @@ public PageInfo(Integer currentPage, Integer pageSize) {
         this.currentPage = currentPage;
     }
 
+    public void setPaginationInfo(Integer total, List<T> totalList) {
+        setTotal(total);
+        setTotalList(totalList);
+        if (total != null && total > 0 && this.pageSize > 0) {
+            this.totalPage = (int) Math.ceil((double) total / this.pageSize);
+        }
+    }
+
```

Comment:
Can you elaborate on why you are doing this?

Reply:
> Can you elaborate on why you are doing this?

Cannot paginate, as described in issue #15606

Reply:
I found there is a more elegent way in #15619. I would prefer merge it. @JohnZp